verticapy/machine_learning/feature_extraction/text.py

```diff
@@ -0,0 +1,197 @@
+from verticapy._typing import SQLRelation
+from verticapy._utils._sql._sys import _executeSQL
+from verticapy.core.vdataframe.base import vDataFrame
+
+class Tfidf:
+    """
+    Create tfidf representation of documents.
+
+    Parameters
+    ----------
+    name: str
+        Name of the model.
+    schema: str
+        Name of the model.
+    overwrite_model: bool, default=False
+        If set to True, training a model with the same
+        name as an existing model overwrites the
+        existing model.
+    lowercase: bool, default=True
+        Converts  all  the elements to lowercase  before
+        processing.
+    """
+    def __init__(
+        self,
+        name: str = None,
+        schema: str = None,
+        overwrite_model: bool = False,
+        lowercase: bool = True
+    ) -> None:
+         self.parameters = {
+             "name": name,
+             "schema": schema,
+             "overwrite_model": overwrite_model,
+             "lowercase": lowercase
+         }
+
+    def fit(
+        self,
+        input_relation: SQLRelation,
+        index: str, 
+        column: str 
+        )-> None:
+        """Apply basic pre-processing. Create table with fitted vocabulary and idf values
+        
+        Parameters
+        ----------
+        input_relation: SQLRelation (db object or vdf)
+        index: str
+            Column name of the document id 
+        column: str
+            Column name which contains the text
+        
+        Returns
+        ----------
+            None    
+        """
+
+        if isinstance(input_relation, vDataFrame):
+            vdf = input_relation.copy()
+        else:
+            vdf = vDataFrame(input_relation)
+        
+        if not self.parameters["lowercase"]:
+            text = {column}
+        else:
+            text = f"LOWER({column})"
+        
+        self.idf_ = ".".join([self.parameters["schema"],self.parameters["name"]])
+
+        if self.parameters["overwrite_model"]:
+            _executeSQL(f"DROP TABLE IF EXISTS {self.idf_}", print_time_sql=False)
+
+        q_idf = f"""CREATE TABLE {self.idf_} AS
```

Comment:
do you really need to create a table? Should it not be another method.
You should probably save your sql somewhere:
```
self.relation = ...
```
and use it later to build your vdf.

Reply:
From my basic testing, creating a table makes the function much faster (4x).
Maybe I could add a parameter to let the user decide if he wants to save the model (pretrained dictionary and idf scores).

Reply:
please give the option, it can be interesting